subxt/src/config/signed_extensions.rs

```diff
@@ -9,42 +9,45 @@
 
 use super::extrinsic_params::{ExtrinsicParams, ExtrinsicParamsEncoder, ExtrinsicParamsError};
 use crate::utils::Era;
-use crate::{client::OfflineClientT, Config, Metadata};
+use crate::{client::OfflineClientT, Config};
 use codec::{Compact, Encode};
 use core::fmt::Debug;
+use derivative::Derivative;
 use scale_decode::DecodeAsType;
-use scale_encode::EncodeAsType;
-use std::marker::PhantomData;
+use scale_info::PortableRegistry;
 
 use std::collections::HashMap;
 
 /// A single [`SignedExtension`] has a unique name, but is otherwise the
 /// same as [`ExtrinsicParams`] in describing how to encode the extra and
 /// additional data.
 pub trait SignedExtension<T: Config>: ExtrinsicParams<T> {
-    /// The name of the signed extension. This is used to associate it
-    /// with the signed extensions that the node is making use of.
-    const NAME: &'static str;
-
     /// The type representing the `extra` bytes of a signed extension.
     /// Decoding from this type should be symmetrical to the respective
     /// `ExtrinsicParamsEncoder::encode_extra_to()` implementation of this signed extension.
     type Decoded: DecodeAsType;
+
+    /// This should return true if the signed extension matches the details given.
+    /// Often, this will involve just checking that the identifier given matches that of the
+    /// extension in question.
+    ///
+    /// The first match that returns true will be the entry that this signed extension
+    /// is used to encode values for. This takes `&mut self`, allowing the extension to
```

Comment:
I don't understand the &mut self part, can you elaborate for my slow eyes?

Reply:
Ah whoops; it took `&mut self` in a prev iteration and I forgot to updatethe comment; good spot!